feat(events): focus event by event bbox

dqunbp opened the pull request:
focus event on click by event geometry bbox
Do not merge until bbox is added to the Event

dqunbp committed:
feat(events): focus event by event bbox

focus event on click by event geometry bbox

Komzpa commented on src/utils/map/cameraForGeometry.ts:
```diff
@@ -3,7 +3,7 @@ import { LAPTOP_WIDTH_PX, MOBILE_WIDTH_PX } from '~utils/hooks/useMediaQuery';
+export function getPaddings(): PaddingOptions {
   const width = window.visualViewport?.width ?? Infinity;
   // mobile
   if (width < MOBILE_WIDTH_PX + 1) return { left: 64, top: 5, right: 115, bottom: 0 };
```
please backlog to calculate this from actual grid

dqunbp replied:
Done
https://kontur.fibery.io/Tasks/My-team's-tasks-225#Task/Calculate-event-bbox-paddings-from-actual-grid-14701

dqunbp committed:
Use real event bbox

propakov commented:
types need to be fixed @dqunbp

propakov commented on src/features/events_list/index.tsx:
```diff
@@ -2,15 +2,17 @@ import { useAction, useAtom } from '@reatom/react';
 
 export function EventList() {
   const [currentEvent, currentEventActions] = useAtom(currentEventAtom);
+  const [, bboxActions] = useAtom(currentEventBbox);
```
maybe useAction would be more suitable?

propakov commented on src/features/events_list/index.tsx:
```diff
@@ -2,15 +2,17 @@ import { useAction, useAtom } from '@reatom/react';
 export function EventList() {
   const [currentEvent, currentEventActions] = useAtom(currentEventAtom);
+  const [, bboxActions] = useAtom(currentEventBbox);
   const scheduleAutoFocus = useAction(scheduledAutoFocus.setTrue);
```
scheduleAutoFocus is no more used here, need to remove

propakov commented on src/core/shared_state/currentEventBbox.ts:
```diff
@@ -0,0 +1,39 @@
+      (event) => event.eventId === currentEvent.id,
+    );
+
+    if (!eventData) throw new Error('Event not found');
```
throwing Error without try catch block is bad (the same. for line 32)
it can break reatom, we had a bug like that recently 
it's better to console.warn to tell developers that something happened, or handle them in try/catch block

propakov commented on src/core/shared_state/currentEventBbox.ts:
```diff
@@ -0,0 +1,39 @@
+import { createAtom } from '@reatom/core';
+import { eventListResourceAtom } from '~features/events_list/atoms/eventListResource';
```
content of core shouldn't import stuff from features as far as I remember (because of circular dependencies)

propakov replied:
it looks like this atom should be replaced to ~features/events_list/atoms/

dqunbp committed:
Resilve review comments

Fix typings
Add precommit typecheck script
Move currentEventBbox atom to event_list feature
Replace throw errors with conole warnings

propakov commented on src/features/events_list/index.tsx:
```diff
@@ -1,16 +1,16 @@
 import { useCallback } from 'react';
 import { currentEventAtom } from '~core/shared_state';
-import { scheduledAutoFocus } from '~core/shared_state/currentEvent';
+import { currentEventBbox } from '~features/events_list/atoms/currentEventBbox';
```
import inside module should be relative  
`import { currentEventBbox } from './atoms/currentEventBbox';`

propakov reviewed (approved):
LGTM, just one import to fix. 👍 

Also a question about .husky/pre-push file in commit. Did you do something with it? 
Just see no changes in it and this info in github 100644 → 100755 says nothing to me :)

propakov reviewed (commented):
It looks like I've found a bug 
Steps:
- go to next event
- refresh screen 
- you see prev event, but you should see last event you've selected, it looks like you need to update url also

https://user-images.githubusercontent.com/20676525/217843481-c17c64fc-995f-47e8-84c6-655ae4efbd0f.mp4

propakov reviewed (changes requested):
need to check/fix this bug before merge

dqunbp committed:
Update url map coords when new event selected

dqunbp commented:
> It looks like I've found a bug Steps:
> 
> * go to next event
> * refresh screen
> * you see prev event, but you should see last event you've selected, it looks like you need to update url also
> 
>  Feb-09-2023.17-37-57.mp4

@propakov fixed

propakov reviewed (approved):
👍
